clients/js/src/getAmountToUiAmount.ts

```diff
@@ -0,0 +1,279 @@
+import {
+  type GetAccountInfoApi,
+  type Rpc,
+  Address,
+  UnixTimestamp,
+  unwrapOption,
+} from '@solana/web3.js';
+import { fetchSysvarClock } from '@solana/sysvars';
+import { fetchMint } from './generated';
+
+/**
+ * Calculates the exponent for the interest rate formula.
+ * @param t1 - The start time in seconds.
+ * @param t2 - The end time in seconds.
+ * @param r  - The interest rate in basis points.
+ *
+ * @returns The calculated exponent.
+ */
+function calculateExponentForTimesAndRate(t1: number, t2: number, r: number) {
+  const ONE_IN_BASIS_POINTS = 10000;
+  const SECONDS_PER_YEAR = 60 * 60 * 24 * 365.24;
+  const timespan = t2 - t1;
+  if (timespan < 0) {
+    throw new Error('Invalid timespan: end time before start time');
+  }
+
+  const numerator = r * timespan;
+  const exponent = numerator / (SECONDS_PER_YEAR * ONE_IN_BASIS_POINTS);
+  return Math.exp(exponent);
+}
+
+/**
+ * Calculates the total scale factor for an interest bearing token by combining two exponential functions:
+ * One for the period between initialization and last update using the pre-update average rate,
+ * and another for the period between last update and current time using the current rate.
+ *
+ * @param currentTimestamp         Current timestamp in seconds
+ * @param lastUpdateTimestamp      Last time the interest rate was updated in seconds
+ * @param initializationTimestamp  Time the interest bearing extension was initialized in seconds
+ * @param preUpdateAverageRate     Interest rate in basis points before last update
+ * @param currentRate              Current interest rate in basis points
+ *
+ * @returns The total scale factor as a product of the two exponential functions
+ */
+function calculateTotalScale({
+  currentTimestamp,
+  lastUpdateTimestamp,
+  initializationTimestamp,
+  preUpdateAverageRate,
+  currentRate,
+}: {
+  currentTimestamp: number;
+  lastUpdateTimestamp: number;
+  initializationTimestamp: number;
+  preUpdateAverageRate: number;
+  currentRate: number;
+}): number {
```

Comment:
I'm not too familiar with the Domain here but in general I think it's better to be verbose when you have many numbers. Otherwise anyone reading `calculateTotalScale(45, 23, 15, 19, 92)` will have no clue what each of these does.

Reply:
Consider me overruled, retracting